crates/cac_client/src/lib.rs

```diff
@@ -204,11 +238,33 @@ impl Client {
         filter_keys: Option<Vec<String>>,
         merge_strategy: MergeStrategy,
     ) -> Result<Map<String, Value>, String> {
-        let mut cac = self.eval(query_data, merge_strategy).await?;
-        if let Some(keys) = filter_keys {
-            cac = filter_keys_by_prefix(cac, keys);
-        }
-        Ok(cac)
+        let filter_string = if let Some(mut vec) = filter_keys.clone() {
+            vec.sort();
+            vec.join(",")
+        } else {
+            "null".to_string()
+        };
+        let hash_key = json_to_sorted_string(&json!(query_data))
+            + "$"
+            + &merge_strategy.clone().to_string()
+            + "$"
+            + &filter_string;
+        let value = self.config_cache.get(&hash_key).unwrap_or({
+            let cac = self.config.read().await;
+            let mut evaled_cac = eval::eval_cac(
+                cac.default_configs.to_owned(),
+                &cac.contexts,
+                &cac.overrides,
+                &query_data,
+                merge_strategy,
+            )?;
```

Comment:
`unwrap_or_else` would not work if you want to use early return from here
in that case you would have to convert this to a `match` block